### DISCUSSION TITLE
Trouble with the map component in th 0.30 version

### DISCUSSION THREAD

**Discussion by SebastiendOrnano** [sqlpage/SQLPage#674] [Q&A] [answered]
I tried to use the map component with the 0.30 version and no way to work : just a message "map loading.". 
I checked everything (including adding leaflet.js, checking cache..) with the simple file from the documentation.
Then I tried again but with an older version (0.19) of sqlpage.bin and it works. 
This component is really great so I hope the problem can be solved.

***and anyway thanks to Ophir and all the developpers for sqlpage !***

**Comment by lovasoa** [comment]
Hello, and thank you for reporting this !

The map component does work on the official site, that uses the latest sqlpage: https://sql.datapage.app/component.sql?component=map

So there must be something particular in your app. Let's debug that together !

Can you please post here : 
 - a `.sql` file that reproduces the issue. Ideally, use only static data. If you need to use data from a database, please post a `.sql` migration file that will allow reproducing the part of your database schema and data needed to reproduce the issue.
 - a screenshot of your browser showing the issue with the web console opened:  right click anywhere on the page, click "Inspect Element", then click "Console", then reload the page and take a screenshot when the error appears.

**Reply by SebastiendOrnano** [reply]
I start with a map.sql linked with a database within a shell. As it was not working I load a very simple map.sql "new delhi" pasted from the documentation but no way. The debug inspection gives an error on the acces to leaflet 
![debugMapSql](https://github.com/user-attachments/assets/ec23c6fd-03ed-4523-8bfc-7ccf652d6919)
 so it is logical not to be able to load the openstreet title.
I tried with another web broswer : chromium with the same result.
I tried with a july version of sqlpage.bin and it woks so I compare the sources of the page and there is a diff on the css (but I don't think it could have an impact) but there also a change on the script 
` <script src="/sqlpage.633f3327e004a42f.js" defer></script>`
against
` <script src="/sqlpage.ae8abf722876661d.js" defer></script>'

I am a no-pro and I am learning sql with my friend iAsk.com who suggests 
`Modifier la Directive CSP : Ajoutez https://cdn.jsdelivr.net à votre directive script-src.` 
 but I feel you are more competent than my friend and you may have another way to unlock the problem. 
Perhaps it is just a question to a csp rule at the server level but It is above my IT skills !

**Reply by lovasoa** [reply]
Ok, interesting. I still cannot reproduce the issue. Do you get the issue with a sql file that literally just contains the following, or do you need a more complicated setup to reproduce the issue ?

```sql
select 
    'map' as component;
select 
    'New Delhi' as title,
    28.6139     as latitude,
    77.209      as longitude;
```

Are you using the card component with the embed property to embed a map, or a custom script, or something in particular in the sqlpage configuration file ? 

Can you reproduce the issue on your computer just by downloading the latest sqlpage version, and pasting the above "Dehli" example into an index.sql file ?

Are you using a particular browser or an old browser version ? Can you reproduce the problem in a private browser window ?

Can you take a screenshot of your full browser window, not just the console ? Ideally the URL, the DOM inspector, and the console would all be visible at the same time, like the following.

![image](https://github.com/user-attachments/assets/0a9a5620-551e-4679-997a-35324f1f4a73)


Any pointer that would help me reproduce your issue would be welcome.

**Reply by lovasoa** [reply]
A screenshot including the network inspector would also be helpful, like the following:

![image](https://github.com/user-attachments/assets/d1967eb1-0fc7-497e-8dc9-07bbb28a70ea)

**Reply by SebastiendOrnano** [reply, marked as answer]
Problem solved !

You said "_Are you using the card component with the embed property to embed a map, or a custom script, or something in particular in the sqlpage configuration file_?"

So I checked everything in my install and I realized that when testing the possibility of coding custom components, I did put a copy of all handlebars in the templates folder so I emptied this folder and it works.

Thank you for your help and sorry for the time you have lost on this issue !

**Reply by lovasoa** [reply]
I'm glad it's solved. Do not hesitate to open new github issues should you encounter other problems.
